### PR TITLE
Don't merge until RC1: Docs for HTTP/3 on IIS

### PR DESCRIPTION
**Pull request by Tratcher**
Contributes to https://github.com/dotnet/aspnetcore/issues/26415

***@Rick-Anderson  edit:***
~Much of this content is the same as Http.Sys (#23252)~
Much of this content is the same as Http.Sys (#23257)

Note this feature work is available in rc2. https://github.com/dotnet/aspnetcore/pull/36054

### CODE REVIEW

**Review by adityamandaleeka — APPROVED** [review]

File: aspnetcore/host-and-deploy/iis/http3.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+The preceding Windows 11 Build versions may require the use of a [Windows Insider](https://insider.windows.com) build.
+
+For an in-process deployment when an HTTP/3 connection is established, [`HttpRequest.Protocol`](xref:Microsoft.AspNetCore.Http.HttpRequest.Protocol*) reports `HTTP/3`. For an out-of-process deployment when an HTTP/3 connection is established, [`HttpRequest.Protocol`](xref:Microsoft.AspNetCore.Http.HttpRequest.Protocol*) reports `HTTP/1.1`.
+
+For more information on the in-process and out-of-process hosting models, see <xref:host-and-deploy/aspnet-core-module>.
+
+HTTP/3 is discovered as an upgrade from HTTP/1.1 or HTTP/2 via the `alt-svc` header. That means the first request will normally use HTTP/1.1 or HTTP/2 before switching to HTTP/3. IIS does not automatically adds the `alt-svc` header, it must be added by the application. Here is a middleware example that adds the `alt-svc` response header, place this early in your request pipeline.
```
Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/iis/http3.md:30]:
```suggestion
HTTP/3 is discovered as an upgrade from HTTP/1.1 or HTTP/2 via the `alt-svc` header. That means the first request will normally use HTTP/1.1 or HTTP/2 before switching to HTTP/3. IIS doesn't automatically add the `alt-svc` header, it must be added by the application. The following code is a middleware example that adds the `alt-svc` response header.
```

File: aspnetcore/host-and-deploy/iis/http3.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
```
Comment (Rick-Anderson) [inline aspnetcore/host-and-deploy/iis/http3.md:31]: Place the preceding code early in the request pipeline.

**Review by Rick-Anderson — APPROVED** [review]

File: aspnetcore/host-and-deploy/iis/http3.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+```C#
+app.Use((context, next) =>
+{
+    context.Response.Headers.AltSvc = "h3=\":443\"";
+    return next(context);
+});
+```
+
```
Comment (Tratcher) [inline aspnetcore/host-and-deploy/iis/http3.md:39]:
```suggestion

Place the preceding code early in the request pipeline.

```

**Comment by Tratcher** [comment]
@Rick-Anderson how do you want to handle the branching for rc2 work?

**Comment by Rick-Anderson** [comment]
I can create a R2 branch when we're ready. But it's not worth the effort to merge this into an RC2 branch.

**Comment by Tratcher** [comment]
On second thought, there was enough working in rc1 to warrant merging this then.

**Pull request merged**